variables.tf

```diff
@@ -0,0 +1,256 @@
+  type        = string
+  description = "(Optional) Set to true to disable rollback of the stack if stack creation failed. Conflicts with `on_failure`."
+  default     = null
+}
+
+variable "notification_arns" {
+  type        = list(string)
+  description = "(Optional) A list of SNS topic ARNs to publish stack related events."
+  default     = []
+}
+
+variable "on_failure" {
+  type        = string
+  description = "(Optional) Action to be taken if stack creation fails. This must be one of: `DO_NOTHING`, `ROLLBACK`, or `DELETE`. Conflicts with `disable_rollback`."
+  default     = null
+}
+
+variable "parameters" {
+  type        = map(string)
+  description = "(Optional) A map of Parameter structures that specify input parameters for the stack."
+  default     = {}
+}
+
+variable "policy_body" {
+  type        = string
+  description = "(Optional) Structure containing the stack policy body. Conflicts with policy_url."
+  default     = null
+}
+
+variable "policy_url" {
+  type        = string
+  description = "(Optional) Location of a file containing the stack policy. Conflicts with `policy_body`."
+  default     = null
+}
+
+variable "tags" {
+  type        = map(string)
+  description = "(Optional) Map of resource tags to associate with this stack. If configured with a provider [`default_tags` configuration block](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/docs#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level."
+  default     = {}
+}
+
+variable "iam_role_arn" {
+  type        = string
+  description = "(Optional) The ARN of an IAM role that AWS CloudFormation assumes to create the stack. If you don't specify a value, AWS CloudFormation uses the role that was previously associated with the stack. If no role is available, AWS CloudFormation uses a temporary session that is generated from your user credentials."
+  default     = null
+}
+
+variable "timeout_in_minutes" {
+  type        = number
+  description = "(Optional) The amount of time that can pass before the stack status becomes `CREATE_FAILED`."
+  default     = null
+}
+
+variable "timeouts" {
+  type        = map(string)
+  description = "configuration block for creating, modifying and destroying stacks"
+  default     = {}
+}
+
+/*
+stack set
+*/
+
+variable "stackset_administration_role_arn" {
+  type        = string
+  description = "(Optional) Amazon Resource Number (ARN) of the IAM Role in the administrator account. This must be defined when using the `SELF_MANAGED` permission model."
+  default     = null
+}
+
+variable "stackset_name" {
+  type        = string
+  description = "(Required) Name of the StackSet. The name must be unique in the region where you create your StackSet. The name can contain only alphanumeric characters (case-sensitive) and hyphens. It must start with an alphabetic character and cannot be longer than 128 characters."
+  default     = "boldlink-test-stackset"
```

Comment:
Is it possible to set this value at the example level?

Reply:
Since the minimal example is not intended to create a stackset, I set the variable the way it is because a `" "` is not a valid default for this variable. Setting it at the example level would be unnecessary as this resource will not be created
@JohnNMwangi